### PR TITLE
PTFE-671 setup mkdocs build

### PR DESCRIPTION
**Pull request by Abubakarr99**

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - scheme: slate
+      toggle:
+        icon: material/brightness-4
+        name: Switch to light mode
+  font:
+    text: Roboto
+    code: Roboto Mono
+
+nav:
+  - Home: index.md
```
Comment (tcarmet) [inline mkdocs.yml:41]: Missing the whole development section

File: pyproject.toml [hunk 2]
```diff
@@ -18,9 +18,12 @@ rq = "^1.15.1"
 pyyaml = "^6.0.1"
 redis = "^4.6.0"
 
+[tool.poetry.group.docs]
+optional = true
```
Comment (tcarmet) [inline pyproject.toml:22]: Ah nice

Check the poetry install on the devcontainer file, and tell it to install all groups of dependencies.

**Review by tcarmet — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (Abubakarr99) [reply pyproject.toml:22]: ok

Comment (Abubakarr99) [reply pyproject.toml:22]: `poetry install --no-interaction --no-ansi --no-root`
it is already installing the dependencies

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -39,7 +39,7 @@
     }
   },
   "remoteUser": "vscode",
-  "postCreateCommand": "poetry install --no-interaction --no-ansi --no-root",
+  "postCreateCommand": "poetry install --no-interaction --no-ansi --no-root --group=docs --group=dev",
```
Comment (tcarmet) [inline .devcontainer/devcontainer.json:42]: Check if there's an install all group option

File: .github/workflows/build-docs.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+          cache: poetry
+      - name: Install Poetry
+        run: |
+          pip install poetry
```
Comment (tcarmet) [inline .github/workflows/build-docs.yaml:20]:
```suggestion
      - name: Install Poetry
        run: |
          pipx install poetry
      - uses: actions/setup-python@v4
        with:
          python-version: 3.x
          cache: poetry
```

**Review by tcarmet — APPROVED** [review]

**Pull request merged**